packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -92,50 +136,31 @@ function BlockInspector() {
 	const blockInspectorAnimationSettings =
 		useBlockInspectorAnimationSettings( blockType );
 
-	const borderPanelLabel = useBorderPanelLabel( {
-		blockName: selectedBlockName,
-	} );
+	const hasSelectedBlocks = selectedBlockCount > 1;
 
-	const hasSelectedBlocks = count > 1;
-
-	if ( hasSelectedBlocks && ! isSectionBlockInSelection ) {
+	if (
+		hasSelectedBlocks &&
+		! isSectionBlockInSelection &&
+		! isSectionBlock
```

Comment:
Isn't it redundant to check whether `! isSectionBlock` as that would already be answered by `! isSectionBlockInSelection`?

Reply:
It's probably a bit edge-casey, but `isSectionBlock` can also refer to a parent of the selected block, so I think that might be why it's included in this check, whereas `isSectionBlockInSelection` is only about the selected blocks?

Reply:
Hmmm, if I'm reading [`isSectionBlock`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/private-selectors.js#L512) correctly, it only checks the block it gets passed. It does look at the section root but only to determine whether that block is a direct child of it (which would mean it's a section). 

I guess ultimately if this addition _does_ change the behaviour, it doesn't belong in a PR that claims to be a straightforward refactor. And if it doesn't, there's no reason to add it 😄

Reply:
> Hmmm, if I'm reading [isSectionBlock](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/private-selectors.js#L512) correctly, it only checks the block it gets passed. It does look at the section root but only to determine whether that block is a direct child of it (which would mean it's a section).

In this case it's a little confusing, I was referring to the `isSectionBlock` const, not the function. Above, `isSectionBlock` is defined by this call:

```
isSectionBlock: _isSectionBlock( renderedBlockClientId ),
```

And it's `renderedBlockClientId` that can be the parent:

https://github.com/WordPress/gutenberg/blob/6e368dd2322d706d020e514219eaf64b1eda3e8b/packages/block-editor/src/components/block-inspector/index.js#L105-L107

Reply:
Oh I see.

Ok so the condition is:

* `hasSelectedBlocks` is true: there's more than one block selected.
* `! isSectionBlockInSelection`: hopefully this one speaks for itself 😅 
* `! isSectionBlock`: if `hasSelectedBlocks` is true, this is always going to be false, because `renderedBlockClientId` is null if there's more than one block selected. 

I think we should remove it for the sake of readability, particularly given this is a refactor meant to improve readability. I'm guessing it's likely to be an AI artifact rather than a human change 😅

Reply:
I'm a bit late to the discussion but if I'm following it correctly, the question is what's the difference between `isSectionBlockInSelection` and `isSectionBlock`?

I think @andrewserong touches on it, that the `renderedBlockClientId` could be the parent section rather than the currently selected block. So this `renderedBlockClientId` could be different than those in the set of currently selected blocks represented by `isSectionBlockInSelection`.

> I'm guessing it's likely to be an AI artifact rather than a human change

Probably more that I messed up the conflict resolution with https://github.com/WordPress/gutenberg/pull/71708 and the PR I had based on this (https://github.com/WordPress/gutenberg/pull/71714).

Reply:
> ! isSectionBlock: if hasSelectedBlocks is true, this is always going to be false, because renderedBlockClientId is null if there's more than one block selected.

Good catch! Yes it sounds like the `! isSectionBlock` in this and the below check are redundant because `isSectionBlock` will always be falsy if `hasSelectedBlocks` is true, like you mention 👍

Reply:
Thanks @aaronrobertshaw! IMO once that's tidied up, this looks in good shape to merge to me.